airbyte-server/src/main/java/io/airbyte/server/handlers/DestinationHandler.java

```diff
@@ -249,6 +272,16 @@ private DestinationRead buildDestinationRead(final UUID destinationId, final Con
     return toDestinationRead(dci, standardDestinationDefinition);
   }
 
+  private DestinationRead buildDestinationReadWithSecrets(final UUID destinationId)
+          throws ConfigNotFoundException, IOException, JsonValidationException {
+
+    // remove secrets from config before returning the read
+    final DestinationConnection dci = Jsons.clone(configRepository.getDestinationConnection(destinationId));
```

Comment:
the object returned here will not have secrets.  but the method name says with secrets. do you mean to use this `getDestinationConnectionWithSecrets`?

Reply:
Got it